vsts/pipelines/_buildParallel.yml

```diff
@@ -0,0 +1,243 @@
+parameters:
+  ascName: OryxMCR
+  acrName: oryxdevmcr.azurecr.io
+
+steps:
+- task: DownloadPipelineArtifact@0
+  displayName: 'Download Pipeline Artifact'
+  inputs:
+    pipelineId: 'Oryx-CI'
+    artifactName: 'drop-$(Build.BuildNumber)'
+    targetPath: '$(Build.SourcesDirectory)'
+  condition: and(eq(variables['Build.DefinitionName'], 'Oryx-CI'), in(variables['Build.Reason'], 'Schedule', 'Manual'))
+
+- script: |
+    if [ "$(BuildBuildImages)" != "true" ] && [ "$(BuildRuntimeImages)" != "true" ] && [ "$(TestIntegration)" != "true" ]
+    then
+      echo "Invalid configuration."
+      echo "Variable 'BuildBuildImages' or 'BuildRuntimeImages' needs to be 'true' to run this build."
+      exit 1
+    fi
+  displayName: 'Validate pipeline run'
+
+- checkout: self
+  clean: true
+
+- task: ShellScript@2
+  displayName: 'Build Oryx.sln'
+  inputs:
+    scriptPath: ./build/build-solution.sh
+
+- task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+  displayName: 'Component Detection - OSS Compliance'
+  inputs:
+    ignoreDirectories: '$(Build.SourcesDirectory)/tests'
+
+- task: Docker@1
+  displayName: Container registry login
+  inputs:
+    command: login
+    azureSubscriptionEndpoint: ${{ parameters.ascName }}
+    azureContainerRegistry: ${{ parameters.acrName }}
+
+- task: ShellScript@2
+  displayName: 'Build build image'
+  inputs:
+    scriptPath: ./build/build-buildimages.sh
+  condition: and(succeeded(), eq(variables['BuildBuildImages'], 'true'))
+
+- task: ShellScript@2
+  displayName: 'Build runtime images'
+  inputs:
+    scriptPath: ./build/build-runtimeimages.sh
+  condition: and(succeeded(), eq(variables['BuildRuntimeImages'], 'true'))
+
+- task: ShellScript@2
+  displayName: 'Test script generator'
+  inputs:
+    scriptPath: ./build/test-buildscriptgenerator.sh
+  condition: and(succeeded(), eq(variables['TestBuildImages'], 'true'))
+
+- task: ShellScript@2
+  displayName: 'Test startup script generators'
+  inputs:
+    scriptPath: ./build/test-startupscriptgenerators.sh
+  condition: and(succeeded(), eq(variables['TestRuntimeImages'], 'true'))
+
+- task: ShellScript@2
+  displayName: 'Test build images'
+  inputs:
+    scriptPath: ./build/test-buildimages.sh
+    args: skipBuildingImages
+  condition: and(succeeded(), eq(variables['TestBuildImages'], 'true'))
+
+- task: ShellScript@2
+  displayName: 'Test runtime images'
+  inputs:
+    scriptPath: ./build/test-runtimeimages.sh
+    args: skipBuildingImages
+  condition: and(succeeded(), eq(variables['TestRuntimeImages'], 'true'))
+
+- task: DownloadBuildArtifacts@0
+  displayName: 'Download Build Artifacts'
+  inputs:
+    artifactName: drop
+  condition: and(succeeded(), eq(variables['TestIntegration'], 'true'))
+
+- task: ms-devlabs.utilitytasks.task-Shellpp.Shell++@0
+  displayName: 'Shell Script'
```

Comment:
What is this for?

Reply:
this is debug, I have several others, I was getting weird test failures so I wanted to see  why .. so you may see some of these, I will make sure I remove this before I try to merge